From 569caac08ecc6c31af68279fcdd54dd88124f49e Mon Sep 17 00:00:00 2001
From: Yousong Zhou <yszhou4tech@gmail.com>
Date: Thu, 2 Feb 2023 14:40:15 +0800
Subject: [PATCH] vpnguru: nft: add ipv6 support

---
 shell/vpnguru/vpnguru | 30 ++++++++++++++++++++++++++----
 1 file changed, 26 insertions(+), 4 deletions(-)

diff --git a/shell/vpnguru/vpnguru b/shell/vpnguru/vpnguru
index 853d338..994f014 100755
--- a/shell/vpnguru/vpnguru
+++ b/shell/vpnguru/vpnguru
@@ -63,17 +63,20 @@ ip() {
 }
 
 vpnguru_nftset_init() {
-	vpnguru_nftset_init_ vpnguru_dst_bypass "$o_dst_bypass $o_remote_servers"
-	vpnguru_nftset_init_ vpnguru_dst_forward "$o_dst_forward"
+	vpnguru_nftset_init_ 4 vpnguru_dst_bypass "$o_dst_bypass $o_remote_servers"
+	vpnguru_nftset_init_ 4 vpnguru_dst_forward "$o_dst_forward"
+	vpnguru_nftset_init_ 6 vpnguru_dst_bypass6 "$o_dst_bypass6 $o_remote_servers6"
+	vpnguru_nftset_init_ 6 vpnguru_dst_forward6 "$o_dst_forward6"
 }
 
 vpnguru_nftset_init_() {
+	local af="$1"; shift
 	local setname="$1"; shift
 	local el
 
 	cat <<EOF
 set $setname {
-	type ipv4_addr;
+	type ipv${af}_addr;
 	flags interval;
 	auto-merge;
 	elements = {
@@ -103,6 +106,8 @@ chain vpnguru_output {
 chain vpnguru_mark {
 	ip daddr @vpnguru_dst_bypass accept;
 	ip daddr @vpnguru_dst_forward meta mark set $o_fwmark;
+	ip6 daddr @vpnguru_dst_bypass6 accept;
+	ip6 daddr @vpnguru_dst_forward6 meta mark set $o_fwmark;
 }
 EOF
 }
@@ -157,11 +162,22 @@ vpnguru_route_init() {
 	ip route flush cache
 }
 
+vpnguru_route6_init() {
+	ip -6 rule add fwmark "$o_fwmark" lookup "$o_rttable"
+	ip -6 route add default dev "$o_ifname" ${o_gateway:+via "$o_gateway6"} table "$o_rttable"
+	ip -6 route flush cache
+}
+
 vpnguru_route_flush() {
-	ip route flush table "$o_rttable"
+	      ip route flush table "$o_rttable"
 	while ip rule del fwmark "$o_fwmark" lookup "$o_rttable" 2>/dev/null; do true; done
 }
 
+vpnguru_route6_flush() {
+	      ip -6 route flush table "$o_rttable"
+	while ip -6 rule del fwmark "$o_fwmark" lookup "$o_rttable" 2>/dev/null; do true; done
+}
+
 arg_ipt=${arg_ipt}
 arg_nft=${arg_nft}
 arg_nft_print_table=${arg_nft_print_table}
@@ -228,6 +244,9 @@ fi
 
 if test -n "$arg_route_do_"; then
 	vpnguru_route_flush
+	if test -n "$arg_nft"; then
+		vpnguru_route6_flush
+	fi
 fi
 
 if test -n "$arg_nft"; then
@@ -265,5 +284,8 @@ fi
 if test -n "$arg_route_do_"; then
 	if test -z "$arg_flush_only"; then
 		vpnguru_route_init
+		if test -n "$arg_nft"; then
+			vpnguru_route6_init
+		fi
 	fi
 fi
